File: src/services/customer/CustomerApi/Models/Customer.cs
namespace CustomerApi.Models;

public class Customer
{
    public Customer() {}

    public Customer(string firstName, string lastName, string email)
    {
        FirstName = firstName;
        LastName = lastName;
        Email = email;
        CreatedAt = DateTime.Now;
    }

    public int Id { get; set; }
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public string Email { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
}

File: src/clients/cli/src/customer.rs
use serde::{
    Serialize,
    Deserialize,
};

/// Holds customer data.
#[derive(Deserialize)]
pub struct CustomerData {
    /// Customer id.
    pub id: i32,
    /// Customer first name.
    #[serde(rename = "firstName")]
    pub first_name: String,
    /// Customer last name.
    #[serde(rename = "lastName")]
    pub last_name: String,
    /// Customer email address.
    pub email: String,
    /// Date and time when created in ISO-8601 format.
    #[serde(rename = "createdAt")]
    pub created_at: String,
}

/// Used to create customers.
#[derive(Serialize)]
pub struct CustomerCreateData {
    /// Customer first name.
    #[serde(rename = "firstName")]
    pub first_name: String,
    /// Customer last name.
    #[serde(rename = "lastName")]
    pub last_name: String,
    /// Customer email address.
    pub email: String,
}

impl CustomerCreateData {
    pub fn new(first_name: &str, last_name: &str, email: &str) -> CustomerCreateData {
        CustomerCreateData {
            first_name: first_name.to_string(),
            last_name: last_name.to_string(),
            email: email.to_string(),
        }
    }
}